src/java.desktop/macosx/classes/com/apple/laf/AquaButtonLabeledUI.java

```diff
@@ -169,7 +169,13 @@ public synchronized void paint(final Graphics g, final JComponent c) {
                 altIcon = b.getIcon();
             }
 
-            altIcon.paintIcon(c, g, iconRect.x, iconRect.y);
+            int offset = 0;
+            if (b.isFocusOwner()) {
+                offset = 2;
+                altIcon = AquaFocus.createFocusedIcon(altIcon, c, 2);
+            }
+
+            altIcon.paintIcon(c, g, iconRect.x - offset, iconRect.y - offset);
```

Comment:
Is it possible that due to offset on the left/top we will draw the icon outside the button bounds on the right/bottom?

Reply:
No. I am substracting the offset to keep drawn image inside the checkbox/radiobutton icon drawing box which is by default size of the icon+4 in width and height, hence i am shifting it up and left to avoid drawing outside the icon box. If size of the image button is artificially limited to the value less or equal of the icon size then halo will be clipped to stay within the borders of the component. In any case we will not overdraw outside of the component.